src/biokbase/narrative/jobs/job.py

```diff
@@ -26,36 +27,75 @@
 
 EXTRA_JOB_STATE_FIELDS = ["batch_id", "cell_id", "run_id"]
 
-JOB_DEFAULTS = {
+JOB_ATTR_DEFAULTS = {
     "app_id": None,
     "app_version": None,
     "batch_id": None,
     "batch_job": False,
     "child_jobs": [],
     "cell_id": None,
-    "extra_data": None,
     "params": None,
-    "owner": None,
+    "retry_ids": [],
+    "retry_parent": None,
     "run_id": None,
     "tag": "release",
+    "user": None,
 }
 
-ALL_JOB_ATTRS = list(JOB_DEFAULTS.keys())
-ALL_JOB_ATTRS.append("job_id")
+JOB_ATTRS = list(JOB_ATTR_DEFAULTS.keys())
+JOB_ATTRS.append("job_id")
+
+# Group attributes by place in nested dict state
+NARR_CELL_INFO_ATTRS = [
+    "cell_id",
+    "run_id",
+    "tag",
+]
+JOB_INPUT_ATTRS = [
+    "app_id",
+    "app_version",
+    "params",
+]
+STATE_ATTRS = list(set(JOB_ATTRS) - set(JOB_INPUT_ATTRS) - set(NARR_CELL_INFO_ATTRS))
+
+
+def _attr_to_ee2(attr):
+    mapping = {
+        "app_version": "service_ver"
+    }
+    if attr in mapping:
+        return mapping[attr]
+    else:
+        return attr
 
 
 class Job(object):
-    app_id = None
-    app_version = None
-    batch_id = None
-    cell_id = None
-    job_id = None
-    params = None
-    run_id = None
     _job_logs = list()
-    _last_state = None
+    _const_state = None  # for original/constant attributes
+    _last_state = None  # for changing attributes
+
+    def __init__(self, ee2_state, extra_data=None, children=None):
+        self._verify_job_id(ee2_state.get("job_id"))
```

Comment:
`_verify_job_id` is only two lines, is it worth making it a separate function?

Reply:
Yeah this can be a pet peeve of mine (unless I do it!)